core/src/main/java/org/hibernate/ogm/dialect/spi/AssociationTypeContext.java

```diff
@@ -20,13 +20,20 @@
 public interface AssociationTypeContext {
 
 	/**
-	 * Provide access to the options set for an entity or property.
+	 * Provide access to the options set for the association.
 	 *
-	 * @return a context object providing access to the options effectively applying for a given entity or property.
+	 * @return a context object providing access to the options effectively applying for the association.
 	 */
 	OptionsContext getOptionsContext();
 
 	/**
+	 * Provide access to the options set for the entity owner of the association.
+	 *
+	 * @return a context object providing access to the options effectively applying for the owner entity.
+	 */
+	OptionsContext getOwnerEntityOptionsContext();
```

Comment:
> What I meant is, can we somehow generalize that, though?

I don't think we can generalize it considering the behavior is going to be different depending on the option we consider.

Typically, you don't want to use the entity options context for options like AssociationStorage.